public/app/features/alerting/unified/components/rule-viewer/tabs/AlertVersionHistory.tsx

```diff
@@ -0,0 +1,188 @@
+          return 0;
+        }
+        return aVersion - bVersion;
+      });
+
+    trackRuleVersionsComparisonClick({
+      latest: newer === ruleVersions[0],
+      oldVersion: older?.grafana_alert.version || 0,
+      newVersion: newer?.grafana_alert.version || 0,
+    });
+
+    setOldVersion(older);
+    setNewVersion(newer);
+    setShowDrawer(true);
+  };
+
+  const handleCheckedVersionChange: ComponentProps<typeof VersionHistoryTable>['onVersionsChecked'] = (versions) => {
+    setCheckedVersions(versions);
+    setOldVersion(undefined);
+    setNewVersion(undefined);
+  };
+
+  /**
+   * Turns a version of a Grafana rule definition into data structure
+   * used to display the version summary when comparing versions
+   */
+  const parseVersionInfo = (version: RulerGrafanaRuleDTO<GrafanaRuleDefinition>): RevisionModel => {
+    const unknown = t('alerting.alertVersionHistory.unknown', 'Unknown');
+    const createdBy = (() => {
+      const updatedBy = version?.grafana_alert.updated_by;
+      const uid = updatedBy?.uid;
+      const name = updatedBy?.name;
+
+      if (!updatedBy) {
+        return unknown;
+      }
+      if (uid && SPECIAL_UID_MAP[uid]) {
+        return SPECIAL_UID_MAP[uid].name;
+      }
+      if (name) {
+        return name;
+      }
+      return uid ? t('alerting.alertVersionHistory.user-id', 'User ID {{uid}}', { uid }) : unknown;
+    })();
+
+    return {
+      created: version.grafana_alert.updated || unknown,
+      createdBy,
+      version: version.grafana_alert.version || unknown,
+    };
+  };
+
+  return (
+    <Stack direction="column" gap={2}>
+      <Text variant="body">
+        <Trans i18nKey="alerting.alertVersionHistory.description">
+          Each time you edit the alert rule a new version is created. You can restore an older version. Select two
+          versions below and compare their differences.
+        </Trans>
+      </Text>
+      <Stack>
+        <Tooltip
+          content={t('core.versionHistory.comparison.select', 'Select two versions to start comparing')}
+          placement="bottom"
+        >
+          <Button type="button" disabled={!canCompare} onClick={compareVersions} icon="code-branch">
+            <Trans i18nKey="alerting.alertVersionHistory.compareVersions">Compare versions</Trans>
+          </Button>
+        </Tooltip>
+      </Stack>
+      {showDrawer && oldVersion && newVersion && (
+        <Drawer
+          onClose={() => setShowDrawer(false)}
+          title={t('alerting.alertVersionHistory.comparing-versions', 'Comparing versions')}
+        >
+          <VersionHistoryComparison
+            oldSummary={parseVersionInfo(oldVersion)}
+            oldVersion={oldVersion}
+            newSummary={parseVersionInfo(newVersion)}
+            newVersion={newVersion}
+            preprocessVersion={preprocessRuleForDiffDisplay}
+          />
+          {config.featureToggles.alertingRuleVersionHistoryRestore && (
```

Comment:
Not functional yet (anything behind the feature toggle can be ignored in this PR)